packages/shared/src/components/modals/CommentUpvotersModal.tsx

```diff
@@ -0,0 +1,69 @@
+import request from 'graphql-request';
+import React, { ReactElement } from 'react';
+import { useInfiniteQuery } from 'react-query';
+import {
+  CommentUpvotesData,
+  COMMENT_UPVOTES_BY_ID_QUERY,
+} from '../../graphql/comments';
+import useFeedInfiniteScroll from '../../hooks/feed/useFeedInfiniteScroll';
+import { apiUrl } from '../../lib/config';
+import { ModalProps } from './StyledModal';
+import { UpvotedPopupModal } from './UpvotedPopupModal';
+
+interface CommentUpvotersModalProps extends ModalProps {
+  commentId: string;
+}
+
+export function CommentUpvotersModal({
+  commentId,
+  ...modalProps
+}: CommentUpvotersModalProps): ReactElement {
+  const commentUpvotesQueryKey = ['commentUpvotes', commentId];
+  const query = useInfiniteQuery<CommentUpvotesData>(
+    commentUpvotesQueryKey,
+    ({ pageParam }) =>
+      request(`${apiUrl}/graphql`, COMMENT_UPVOTES_BY_ID_QUERY, {
+        id: commentId,
+        first: 10,
+        after: pageParam,
+      }),
+    {
+      enabled: !!commentId,
+      getNextPageParam: (lastPage) =>
+        lastPage.commentUpvotes.pageInfo.hasNextPage &&
+        lastPage.commentUpvotes.pageInfo.endCursor,
+    },
+  );
+
+  const upvotes = React.useMemo(
```

Comment:
It's better to make the modal compatible with the pagination object rather than flattening the array I think. All our use cases depend on a pagination object. Should be better in terms of perf

Reply:
I was thinking to make another Modal right now, that will be able incorporate the `useInfiniteScroll`. While I was typing my thoughts (which I have deleted now) it came to my mind that yes, the `key` attribute would help a lot in terms of rendering purposes.